src/DotNet/ReflectionExtensions.cs

```diff
@@ -105,5 +105,11 @@ static class ReflectionExtensions {
 		/// <param name="type">this</param>
 		public static bool IsTypeDef(this Type type) =>
 			!(type is null) && !type.HasElementType && (!type.IsGenericType || type.IsGenericTypeDefinition);
+
+		internal static string Unescape(string name) {
+			if (string.IsNullOrEmpty(name))
+				return name;
+			return name.Replace(@"\,", ",").Replace(@"\+", "+").Replace(@"\&", "&").Replace(@"\*", "*").Replace(@"\[", "[").Replace(@"\]", "]").Replace(@"\\", @"\");
```

Comment:
This code is inefficient. You should have a check to see if it has a `\` and if not, return the input string. If there's a `\`, use a string builder.